### DISCUSSION TITLE
Binary Sensor Type value 0xFF not handled correctly

### DISCUSSION THREAD

**Discussion by spudwebb** [zwave-js/zwave-js#6923] [Request Support / Investigate Issue] [answered]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

According to the ZWave specs for the Binary Sensor CC version 2, when receiving a binary sensor type value 0xFF in a binary sensor report we should look at the first supported sensor type to determine the actual sensor type of the report, instead ZWave JS just use the "Any" type and create an "Any" value instead of the actual type value.

> Sensor Type (8 bits)
> Sensor Type specifies what type of sensor this command originates from. Refer to Table 2.92. The
> sensor type value 0xFF returns the first found supported sensor type in the bit mask (starting from
> bit 0 in Bit Mask 1) by the Binary Sensor Supported Report.

See for example this Ecolink Motion sensor. During interview it correctly reports that the only binary sensor type supported is Motion, but then when queried for its value, it reports a sensor type=0xFF which is then translated by ZWaveJS as Any instead of Motion.

```
09:43:26.441 CNTRLR   [Node 084] Interviewing Binary Sensor...

09:43:26.441 CNTRLR » [Node 084] querying supported sensor types...

09:43:26.443 DRIVER » [Node 084] [REQ] [SendData]
                      │ transmit options: 0x25
                      │ callback id:      67
                      └─[BinarySensorCCSupportedGet]

09:43:26.599 CNTRLR   [Node 084] [+] [Binary Sensor] supportedSensorTypes: 1 [Endpoint 0] [internal]
                      2

09:43:26.599 DRIVER « [Node 084] [REQ] [ApplicationCommand]
                      └─[BinarySensorCCSupportedReport]
                          supported types: Motion

09:43:26.600 CNTRLR « [Node 084] received supported sensor types: 
                      · Motion

09:43:26.600 CNTRLR » [Node 084] querying current value for Motion...

09:43:26.602 DRIVER » [Node 084] [REQ] [SendData]
                      │ transmit options: 0x25
                      │ callback id:      68
                      └─[BinarySensorCCGet]
                          type: Motion

09:43:26.809 CNTRLR   [Node 084] [Binary Sensor] Any: metadata updated                  [Endpoint 0]

2024-06-12 09:43:26.809 INFO Z-WAVE: [Node 084] Metadata updated: 48-0-Any

09:43:26.810 CNTRLR   [Node 084] [+] [Binary Sensor] Any: false                         [Endpoint 0]

2024-06-12 09:43:26.810 INFO Z-WAVE: [Node 084] Value added: 48-0-Any => false

09:43:26.810 DRIVER « [Node 084] [REQ] [ApplicationCommand]
                      └─[BinarySensorCCReport]
                          type:  Any
                          value: false
```
Sorry I don't have a full debug log as I don't own this device myself, the log is coming from another user.

### Software versions

Driver (node-zwave-js): 12.9.1

Z-Wave JS UI: 9.13.3

### Device information

Manufacturer: Ecolink
Model name: PIR Motion Sensor v2.5
Node ID: 84

### Checklist

- [x] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [X] The log includes a re-interview of the problematic device _(if applicable)_.

- [X] The log includes the problematic interaction with the device _(if applicable)_.

- [X] I provided the node ID of the problematic device _(if applicable)_.

### Upload Logfile

[Node 084 Log.txt](https://github.com/user-attachments/files/15820731/Node.084.Log.txt)

**Comment by AlCalzone** [comment, marked as answer]
Nice find! Escalated to an issue: https://github.com/zwave-js/node-zwave-js/issues/6924
